### PR TITLE
Fixed 4728 : Suggest User to turn on camera location

### LINKED ISSUE #4728: Suggest to users to turn on camera location.
For people whose 10 last uploads had no EXIF coordinates (latitude, longitude): show this tip activity just after they push the "Submit" button of the last step of the Upload Wizard:

>Location data helps Wiki editors find your picture, making it much more useful.
>Your recent uploads have no location.
>We suggest you **turn on location** in your camera app's settings.
>Thank you for uploading!

Showing the tip should reset the counter, meaning the tip will appear again after 10 more consecutive pictures without location.

That's important because many camera apps have location off by default.

### PR DESCRIPTION
**Pull request by Ayan-10**
**Description (required)**

Fixes #4728 

Suggested user to turn on camera location after 10 consecutive uploads with no location.

**Tests performed (required)**

Tested latest master on Pixel 3 with API level 30.

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java [hunk 1]
```diff
@@ -469,7 +471,18 @@ public void onNextButtonClicked(int index) {
             vpUpload.setCurrentItem(index + 1, false);
             fragments.get(index + 1).onBecameVisible();
         } else {
-            presenter.handleSubmit();
+            if(defaultKvStore.getInt(COUNTER_OF_NO_LOCATION, 0) == 10){
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:474]: Would you mind changing `==10` to `>= 10`?
It is a usual pattern, for safety.

**Review by nicolas-raoul — COMMENTED** [review]
Are you counting 10 pictures uploaded, or 10 uses of the Upload Wizard? (for instance I can use the Upload Wizard to upload 3 or even 20 pictures)
Any of the two is fine, actually I had not thought about this when writing the issue.
Thanks!

**Comment by Ayan-10** [comment]
@nicolas-raoul It will check for every contribution/picture whether it has coordinates or not. If 10 consecutive pictures/contributions do not have coordinates. It will show the message.

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java (hunk 1 above)
Comment (Ayan-10) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java]: Sure

File: app/src/main/java/fr/free/nrw/commons/upload/UploadPresenter.java [hunk 2]
```diff
@@ -32,6 +34,9 @@
     private UploadContract.View view = DUMMY;
 
     private CompositeDisposable compositeDisposable;
+    public static final String COUNTER_OF_NO_LOCATION
+        = "how_many_consecutive_uploads_have_no_coordinates";
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadPresenter.java:38]: @misaochan Do we have a convention for these store key values? I think there was an issue about that, but can't find it.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -678,4 +678,6 @@ Upload your first media by tapping on the add button.</string>
   <string name="advanced_query_info_text">You can customize the Nearby query. If you get errors, reset and apply.</string>
   <string name="apply">Apply</string>
   <string name="reset">Reset</string>
+  <string name="turn_on_camera_location_title">Turn on location in your camera app\'s settings</string>
```
Comment (nicolas-raoul) [inline app/src/main/res/values/strings.xml:681]: Since it is just a title, could you please change this to just `Camera location`? Thanks!

Comment (nicolas-raoul) [reply app/src/main/res/values/strings.xml:681]: Or if it is possible to not put any title, that would be even better. The message below is already enough, I feel.

File: app/src/main/java/fr/free/nrw/commons/upload/UploadPresenter.java (hunk 2 above)
Comment (misaochan) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadPresenter.java:38]: Not as far as I'm aware, @neslihanturan any idea?

Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadPresenter.java:38]: @Ayan-10 The feature works great, and the code looks fine. Could you please just rename `COUNTER_OF_NO_LOCATION` to `COUNTER_OF_CONSECUTIVE_UPLOADS_WITHOUT_COORDINATES` (sorry a bit long but explicit) and `how_many_consecutive_uploads_have_no_coordinates` to `number_of_consecutive_uploads_without_coordinates`, if it sounds good to you? Thanks!

Comment (Ayan-10) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadPresenter.java:38]: That looks more accurate. Sure, I will change it.

**Review by nicolas-raoul — APPROVED** [review]

**Comment by nicolas-raoul** [comment]
Thanks @Ayan-10, great enhancement!

**Pull request merged**